democracy/views/hearing.py

```diff
@@ -99,8 +100,25 @@ def get_serializer(self, *args, **kwargs):
         kwargs['context'] = self.get_serializer_context()
         return serializer_class(*args, **kwargs)
 
+    @staticmethod
+    def check_preview_code(id, code):
```

Comment:
The key-generating logic should be refactored into a function that can be shared between here and the model.